packages/synapse-react-client/src/components/ChallengeTeamWizard/CreateChallengeTeam.test.tsx

```diff
@@ -0,0 +1,217 @@
+    mockCreateAndRegisterTeam.mockResolvedValue(MUTATION_SUCCESS_RESULT)
+
+    const {
+      ref,
+      user,
+      nameField,
+      descriptionField,
+      messageField,
+      inviteeField,
+      mockOnCanSubmitChange,
+      mockOnFinished,
+    } = renderComponent()
+
+    await user.type(nameField, TEAM_NAME)
+    await user.type(descriptionField, TEAM_DESCRIPTION)
+    await user.type(messageField, RECRUITMENT_MESSAGE)
+    await user.type(inviteeField, EMAILS_TO_SEND.join(', '))
+
+    // Verify that we can submit the form
+    expect(mockOnCanSubmitChange).toHaveBeenLastCalledWith(true)
+
+    // Verify the form has not been submitted yet
+    expect(mockCreateAndRegisterTeam).not.toHaveBeenCalled()
+    expect(mockOnFinished).not.toHaveBeenCalled()
+
+    // Submit the form
+    act(() => {
+      ref.current!.submit()
+    })
+
+    // Verify that the data was passed to the hook, and onFinished was called
+    await waitFor(() => {
+      expect(mockCreateAndRegisterTeam).toHaveBeenCalledTimes(1)
+      expect(mockCreateAndRegisterTeam).toHaveBeenCalledWith(
+        {
+          name: TEAM_NAME,
+          description: TEAM_DESCRIPTION,
+        },
+        MOCK_CHALLENGE_ID,
+        EMAILS_TO_SEND,
+        RECRUITMENT_MESSAGE,
+      )
+      expect(mockOnFinished).toHaveBeenCalledTimes(1)
+    })
+  })
+
+  it('Prevents submitting the form before a team name is filled in', async () => {
+    const { user, nameField, mockOnCanSubmitChange } = renderComponent()
+
+    // Verify that the form cannot be submitted
+    expect(mockOnCanSubmitChange).toHaveBeenLastCalledWith(false)
+
+    await user.type(nameField, TEAM_NAME)
+
+    // Verify that we can submit the form
+    expect(mockOnCanSubmitChange).toHaveBeenLastCalledWith(true)
+  })
+
+  it('Does not permit inviting more than 3 email addresses', async () => {
+    const { user, nameField, inviteeField, mockOnCanSubmitChange } =
+      renderComponent()
+
+    await user.type(nameField, TEAM_NAME)
+    await user.type(inviteeField, EMAILS_TO_SEND.join(', '))
+
+    // Verify that we can submit the form with 3 email addresses
+    expect(mockOnCanSubmitChange).toHaveBeenLastCalledWith(true)
+
+    // Add another and verify that an error appears
+    await user.type(inviteeField, ', tooMany@emails.net')
+    const alert = await screen.findByRole('alert')
+    within(alert).getByText(TOO_MANY_INVITEES_ERROR)
+  })
+
+  it('Does not call onFinished when the mutation returns an error', async () => {
+    const errorMessage = 'A team with that name already exists, so pick another'
+    const error = new SynapseClientError(
+      400,
+      errorMessage,
+      expect.getState().currentTestName!,
+    )
+    // TODO: Update the hook result to return an error AFTER mockCreateAndRegisterTeam is called
+    mockUseCreateAndRegisterTeam.mockReturnValue({
+      createAndRegisterTeam: mockCreateAndRegisterTeam,
+      isPending: false,
+      errors: [error],
+    })
```

Comment:
I couldn't get the React component to see the `errors` hook result change after the initial render, so I just had the test initially return the `errors` that we expect.